tests/mr/device/mr_ref_tests.cpp

```diff
@@ -33,6 +33,7 @@ INSTANTIATE_TEST_SUITE_P(ResourceTests,
 #endif
                                            mr_factory{"Managed", &make_managed},
                                            mr_factory{"Pool", &make_pool},
+                                           mr_factory{"HostPinnedPool", &make_host_pinned_pool},
```

Comment:
**💡 suggestion:** ‏ 

```suggestion
                                           mr_factory{"HostPinned", &make_host_pinned},
                                           mr_factory{"HostPinnedPool", &make_host_pinned_pool},
```

Reply:
This is actually not possible with the current test setup. 

The tests machinery currently requires a `std::function<std::shared_ptr<rmm::mr::device_memory_resource>>`. However, `pinned_host_memory_resource` is not derived from `device_memory_resource` so that wont compile.

I am wary of completely rewriting the test infra in this PR.

Reply:
Oh I thought when you created device_mr_ref_tests that you had changed that machinery for the ref tests. OK, that can be a follow up.

Reply:
The issue is that we need to store the resource somewhere, as `resource_ref` is only a reference and now owning.

That is why I kept most of the base machinery unchanged

Reply:
OK. Seems we need to find a way to test both the (new) MR(s) that just implement the concepts and the (legacy) MRs that just derive from device_memory_resource.

Reply:
@miscco did you have further thoughts here?

> Seems we need to find a way to test both the (new) MR(s) that just implement the concepts and the (legacy) MRs that just derive from device_memory_resource.